File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/Human.java
package model.Organisms;

import model.Animal;
import model.Organism;
import model.Point;
import model.World;

import java.awt.*;

public class Human extends Animal {
    private Color humanColor = Color.PINK;
    private boolean immortality = false;
    private int imortalityRounds = 0;

    public Human(World world, Point position){
        super(world, position, 5, 4);
    }


    public boolean getImmortality() {
        return immortality;
    }

    public int getImortalityRounds(){
        return imortalityRounds;
    }

    public void setImortalityRounds(int tmp){
        imortalityRounds = tmp;
    }

    public void setImmortality(boolean tmp) {
        this.immortality = tmp;
    }

    @Override
    public Color getColor(){
        return humanColor;
    }

    @Override
    public String getNazwa(){
        return "Czlowiek";
    }

    @Override
    public Organism makeNewOrganism(Point position){
        return new Human(world, position);
    }

    private void immortalityFunction(){
        if (getImmortality()){
            if (--imortalityRounds <= 0){
                setImmortality(false);
            }
        }
    }

    @Override
    public void Action(){
        increaseAge();

        if (world.humanMove != null){
            Point currentPos = getPosition();
            World.Direction direction = world.humanMove;
            Point newPosition = new Point(currentPos.getX(), currentPos.getY());

            switch (direction){
                case GORA -> newPosition.setY(currentPos.getY() - 1);
                case DOL -> newPosition.setY(currentPos.getY() + 1);
                case LEWO -> newPosition.setX(currentPos.getX() - 1);
                case PRAWO -> newPosition.setX(currentPos.getX() + 1);
            }

            world.tryToMoveOrganism(this, newPosition);
            world.humanMove = null;
        }

        immortalityFunction();
    }

    @Override
    public void Collision(Organism attacker){
        if (getImmortality()) {
            world.addEvent(this.getNazwa() + " uniknął śmierci dzięki nieśmiertelności przed " + attacker.getNazwa() + " na pozycji (" + getPosition().getX() + ", " + getPosition().getY() + ")");
            attacker.setAlive(false);
            return;
        }

        if(attacker.getPower() >= this.getPower()){
            world.killHuman();
            if (!getIsAlive()){
                world.addEvent(attacker.getNazwa() + " zabił " + this.getNazwa() + " na pozycji (" + getPosition().getX() + ", " + getPosition().getY() + ")");
                Point newAttackerPos = new Point(getPosition().getX(), getPosition().getY());
                world.moveOrganism(attacker, newAttackerPos);
            }
        } else {
            attacker.setAlive(false);
            world.addEvent(this.getNazwa() + " zabił " + attacker.getNazwa() + " na pozycji (" + getPosition().getX() + ", " + getPosition().getY() + ")");
            Point newDeffenderPos = new Point(getPosition().getX(), getPosition().getY());
            if (this.getIsAlive()){
                world.moveOrganism(this, newDeffenderPos);
            }
        }
    }

}

File: WirtualnySwiatPython/Animals/Human.py
from animal import Animal
from config import PINK, GORA, DOL, PRAWO, LEWO
from point import Point


class Human(Animal):
    def __init__(self, world, position):
        super().__init__(world, position, power=5, initiative=4)
        self.color = PINK
        self.immoratality = False
        self.immoratalityRounds = 0

    def getColor(self):
        return self.color

    def getNazwa(self):
        return "Czlowiek"

    def setImmortality(self, imo):
        self.immoratality = imo

    def getImmortality(self):
        return self.immoratality

    def setImmortalityRounds(self, tmp):
        self.immoratalityRounds = tmp

    def getImmortalityRounds(self):
        return self.immoratalityRounds

    def makeNewOrganism(self, position):
        return Human(self.world, position)

    def immortalityFunction(self):
        if self.immoratality:
            self.immoratalityRounds -= 1
            if self.immoratalityRounds < 0:
                self.immoratality = False

    def Action(self):
        self.increaseAge()

        if self.world.humanMove is not None:
            position = self.getPosition()
            direction = self.world.humanMove

            new_x = position.GetX()
            new_y = position.GetY()

            if direction == GORA:
                new_y -= 1
            elif direction == PRAWO:
                new_x += 1
            elif direction == DOL:
                new_y += 1
            elif direction == LEWO:
                new_x -= 1

            newPosition = Point(new_x, new_y)
            self.world.tryToMoveOrganism(self, newPosition)
            self.world.humanMove = None

        self.immortalityFunction()